vumi/persist/fields.py

```diff
@@ -56,24 +57,52 @@ def _add_index(self, modelobj, value):
             return
         modelobj._riak_object.add_index(self.index_name, str(value))
 
+    def get_riak_data(self, modelobj, default=None, key=None):
+        if key is None:
+            key = self.key
+        return modelobj._riak_object.get_data().get(key, default)
+
+    def set_riak_data(self, modelobj, raw_value, key=None):
+        if key is None:
+            key = self.key
+        old_raw_value = modelobj._riak_object.get_data().get(key)
+        # We set this even if it's "unmodified" because we want explicit `None`
+        # values rather than missing fields.
+        modelobj._riak_object.set_data_field(key, raw_value)
```

Comment:
Makes me sad that we can't do any optimisation to save writes (even by special-casing `None`s?). Otoh, we don't know the performance benefit so I probably shouldn't worry so much...